legend-engine-xt-persistence-test-runner/src/main/java/org/finos/legend/engine/testable/persistence/extension/PersistenceTestRunner.java

```diff
@@ -107,7 +125,17 @@ public TestResult executeAtomicTest(Root_meta_pure_test_AtomicTest atomicTest, P
                     {
                         // Retrieve testData
                         String testDataString = getConnectionTestData(testBatch.testData);
-                        invokePersistence(targetDataset, persistence, testDataString, connection);
+
+                        // Test runner flow for v2
+                        if (!isPersistenceSpecModelV1 && serviceOutputTarget != null)
```

Comment:
It is valid for `serviceOutputTarget` to be null when using v2?  

Seems odd that we could be on v2, but if `serviceOutputTarget` is null, we execute the v1 flow.

Reply:
this was just to make sure that there is no spec without a serviceOutputTarget block. Will refactor this to include a validate() function to make sure user doesn't pass a spec without persister or serviceOutputTarget blocks.

Reply:
Still, lets make the if conditions here clearer and fail if preconditions are not met so we dont depend on some other code outside and easy to follow for anyone reading the code.

Reply:
I've made changes in the code for a better flow - could you take a look at line 89  validatePersistenceSpec(persistence) 
and changes to the conditions ( lines 132 - 140)